### PR TITLE
REP-3270 Rackspace Auth User filter wrapper updates

### PR DESCRIPTION
**Pull request by Mario-Lopez**
Updated filter:
1. away from the old filter pattern
2. away from using the old and busted wrappers

I kept the handler class and updated its interface away from the old pattern since it had a lot of logic and more importantly unit tests.  I moved a bit of the logic into the Filter class since I thought it made more sense there.

The design structure isn't the way I would have written it from scratch, but I was able to make relatively few changes to accomplish our goals.

### CODE REVIEW

**Comment by wdschei** [comment]
The code changes look good, but it seems to have failed 20+ functional tests. We'll see what @jennyvo has to say during her review. Maybe they are just needed updates to the tests, but I don't think they are.

**Comment by jennyvo** [comment]
will take a look those tests failed

**Comment by jennyvo** [comment]
wdschei, Mario-Lopez, I checked test failed for real, the filter is used to extract username and group to x-pp-user and x-pp-groups from auth user payload to rate limiting purposes

**Comment by wdschei** [comment]
That is what I was afraid of at first glance.

**Comment by jennyvo** [comment]
debugged through seeing that mutableHttpRequest added header but username as key and group name as value

File: repose-aggregator/components/filters/rackspace-auth-user/src/main/scala/org/openrepose/filters/rackspaceauthuser/RackspaceAuthUserFilter.scala [hunk 1]
```diff
@@ -19,40 +19,74 @@
...
+    }
+
+    val httpServletRequest = servletRequest.asInstanceOf[HttpServletRequest]
+    if (httpServletRequest.getMethod != "POST") {
+      // this filter only operates on POST requests that have a body to parse
+      filterChain.doFilter(servletRequest, servletResponse)
+    } else {
+      val wrappedRequest = new HttpServletRequestWrapper(httpServletRequest)
+      handler.get.parseUserGroupFromInputStream(wrappedRequest.getInputStream, wrappedRequest.getContentType) foreach { rackspaceAuthUserGroup =>
+        wrappedRequest.addHeader(rackspaceAuthUserGroup.user, rackspaceAuthUserGroup.group, rackspaceAuthUserGroup.quality)
```
Comment (jennyvo) [inline repose-aggregator/components/filters/rackspace-auth-user/src/main/scala/org/openrepose/filters/rackspaceauthuser/RackspaceAuthUserFilter.scala]: added header but username as name and group name as value?

Comment (Mario-Lopez) [inline repose-aggregator/components/filters/rackspace-auth-user/src/main/scala/org/openrepose/filters/rackspaceauthuser/RackspaceAuthUserFilter.scala]: D'oh!  Thanks for finding this!  I pushed a fix.

**Comment by Mario-Lopez** [comment]
Correct number of functional test failures now.

**Comment by wdschei** [comment]
Yea. Only 27 tests failed!!! Something just doesn't seem right about saying that.

**Pull request merged**